frontend/packages/dev-console/src/components/health-checks/AddHealthChecks.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form, Button } from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+import {
+  ContainerDropdown,
+  history,
+  PageHeading,
+  ResourceLink,
+  ResourceIcon,
+} from '@console/internal/components/utils';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { FormFooter } from '@console/shared';
+import { getResourcesType } from '../edit-application/edit-application-utils';
+import HealthChecks from './HealthChecks';
+import Helmet from 'react-helmet';
+import { ContainerModel } from '@console/internal/models';
+import './AddHealthChecks.scss';
+import { getHealthChecksData } from './create-health-checks-probe-utils';
+
+type AddHealthChecksProps = {
+  resource?: K8sResourceKind;
+  currentContainer: string;
+};
+
+const AddHealthChecks: React.FC<FormikProps<FormikValues> & AddHealthChecksProps> = ({
+  resource,
+  currentContainer,
+  handleSubmit,
+  handleReset,
+  dirty,
+  errors,
+  status,
+  isSubmitting,
+  setFieldValue,
+}) => {
+  const [currentKey, setCurrentKey] = React.useState(currentContainer);
+  const containers = resource?.spec?.template?.spec?.containers;
+  const healthCheckAdded = _.every(
+    containers,
+    (container) => container.readinessProbe || container.livenessProbe || container.startupProbe,
+  );
+  const containersByKey = _.keyBy(containers, 'name');
+  const pageTitle = healthCheckAdded ? 'Edit Health Checks' : 'Add Health Checks';
+
+  const handleSelectContainer = (containerName: string) => {
+    const containerIndex = _.findIndex(resource.spec.template.spec.containers, [
+      'name',
+      containerName,
+    ]);
+    setCurrentKey(containerName);
+    setFieldValue('containerName', containerName);
+    setFieldValue('healthChecks', getHealthChecksData(resource, containerIndex));
+    history.push(
+      `/k8s/ns/${resource.metadata.namespace}/${resource.kind}/${resource.metadata.name}/containers/${containerName}/health-checks`,
```

Comment:
URL cannot rely on kind alone.
I think you should replace the URL instead of push. That way on submit / reset you can go back in history. 
Currently switching containers and pressing cancel gives an odd experience.

Reply:
replaced the url in history instead of push.